pkg/kubelet/util/util_windows.go

```diff
@@ -71,12 +72,12 @@ func GetAddressAndDialer(endpoint string) (string, func(addr string, timeout tim
 	return "", nil, fmt.Errorf("only support tcp and npipe endpoint")
 }
 
-func tcpDial(addr string, timeout time.Duration) (net.Conn, error) {
-	return net.DialTimeout(tcpProtocol, addr, timeout)
+func tcpDial(ctx context.Context, addr string) (net.Conn, error) {
+	return net.DialTCP(tcpProtocol, nil, &net.TCPAddr{IP: net.ParseIP(addr)})
```

Comment:
why is discarding the deadline in the context ok here? I expected something more like:

```go
return (&net.Dialer{}).DialContext(ctx, tcpProtocol, addr)
```